gluten-core/src/main/scala/org/apache/gluten/extension/columnar/TransformHintRule.scala

```diff
@@ -416,16 +387,7 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
           val transformer = ColumnarShuffleExchangeExec(plan, plan.child, plan.child.output)
           transformer.doValidate().tagOnFallback(plan)
         case plan: ShuffledHashJoinExec =>
-          val transformer = BackendsApiManager.getSparkPlanExecApiInstance
-            .genShuffledHashJoinExecTransformer(
-              plan.leftKeys,
-              plan.rightKeys,
-              plan.joinType,
-              TransformHints.getShuffleHashJoinBuildSide(plan),
```

Comment:
It seems we have already chosen a small table before. The only thing different with before is, we convert smj to shj from strategy to columnar, so is GlutenCostEvaluator required ? I think the cost evaluator is useful when we add tag in strategy.

Reply:
The size obtained here may not be accurate. If `currentPhysicalPlan == newPhysicalPlan`, logicalLink will not be updated. GlutenCostEvaluator is to update logicalplan when `currentPhysicalPlan == newPhysicalPlan`, and also update sizeByte.